Configuration/Skimming/src/LeptonSkimming.cc

```diff
@@ -0,0 +1,487 @@
+ Probee_Cut=runParameters.getParameter<double>("Probee_Cut");
+ Cosee_Cut=runParameters.getParameter<double>("Cosee_Cut");
+ 
+ PtKTrack_Cut=runParameters.getParameter<double>("PtKTrack_Cut");
+ MaxMB_Cut=runParameters.getParameter<double>("MaxMB_Cut");
+ MinMB_Cut=runParameters.getParameter<double>("MinMB_Cut");
+ TrkTrkMinDR_Cut=runParameters.getParameter<double>("TrkTrkMinDR_Cut");
+
+ TrackSdxy_Cut=runParameters.getParameter<double>("TrackSdxy_Cut");
+ 
+ MuTrgMatchCone=runParameters.getParameter<double>("MuTrgMatchCone");
+ SkipIfNoMuMatch=runParameters.getParameter<bool>("SkipIfNoMuMatch");
+ EpairZvtx_Cut=runParameters.getParameter<double>("EpairZvtx_Cut");
+ Ksdxy_Cut=runParameters.getParameter<double>("Ksdxy_Cut");
+ ProbeeK_Cut=runParameters.getParameter<double>("ProbeeK_Cut");
+ CoseeK_Cut=runParameters.getParameter<double>("CoseeK_Cut");
+ TrackMuDz_Cut=runParameters.getParameter<double>("TrackMuDz_Cut");
+ MaxMVA_Cut=runParameters.getParameter<double>("MaxMVA_Cut");
+ MinMVA_Cut=runParameters.getParameter<double>("MinMVA_Cut");
+ TrgExclusionCone=runParameters.getParameter<double>("TrgExclusionCone");
+ SLxy_Cut=runParameters.getParameter<double>("SLxy_Cut");
+ PtB_Cut=runParameters.getParameter<double>("PtB_Cut");
+ PtMu_Cut=runParameters.getParameter<double>("PtMu_Cut");
+  PtEl_Cut=runParameters.getParameter<double>("PtEl_Cut");
+ QualMu_Cut=runParameters.getParameter<double>("QualMu_Cut");
+ MuTrgExclusionCone=runParameters.getParameter<double>("MuTrgExclusionCone");
+ ElTrgExclusionCone=runParameters.getParameter<double>("ElTrgExclusionCone");
+ TrkObjExclusionCone=runParameters.getParameter<double>("TrkObjExclusionCone");
+ MuTrgMuDz_Cut=runParameters.getParameter<double>("MuTrgMuDz_Cut");
+ ElTrgMuDz_Cut=runParameters.getParameter<double>("ElTrgMuDz_Cut");
+ ObjPtLargerThanTrack=runParameters.getParameter<bool>("ObjPtLargerThanTrack");
+
+
+
+}
+
+
+LeptonSkimming::~LeptonSkimming()
+{
+ 
+   // do anything here that needs to be done at destruction time
+   // (e.g. close files, deallocate resources etc.)
+
+}
+
+bool LeptonSkimming::sort_by_MaxPt(const std::vector<float>& a, const std::vector<float>& b) {
+  return a[0] > b[0];
+  }
+
+bool LeptonSkimming::HLTFired(const edm::Event& iEvent, const edm::EventSetup& iSetup,std::vector<string> HLTPath ){
+  using namespace std;  using namespace edm;  using namespace reco;
+  using namespace trigger;
+ 
+  edm::Handle<edm::TriggerResults> trigResults;
+  iEvent.getByToken(trgresultsToken_, trigResults);
+ 
+  
+  bool fire= false;
+  if( trigResults.failedToGet() ) return false;
+  for (unsigned int ip=0; ip<HLTPath.size(); ip++){
+  int N_Triggers = trigResults->size();
+  const edm::TriggerNames & trigName = iEvent.triggerNames(*trigResults);    
+  //cout << "new" << endl;
+  for( int i_Trig = 0; i_Trig < N_Triggers; ++i_Trig ) {
+    std::string TrigPath =trigName.triggerName(i_Trig); 
+    if (!trigResults->accept(i_Trig))   continue;
```

Comment:
better to move this line up one line to avoid unneeded calculations

Reply:
I agree, I did it